xarray/testing.py

```diff
@@ -60,13 +61,18 @@ def assert_equal(a, b):
     numpy.testing.assert_array_equal
     """
     __tracebackhide__ = True
-    assert type(a) == type(b)
-    if isinstance(a, (Variable, DataArray)):
-        assert a.equals(b), formatting.diff_array_repr(a, b, "equals")
-    elif isinstance(a, Dataset):
-        assert a.equals(b), formatting.diff_dataset_repr(a, b, "equals")
-    else:
-        raise TypeError("{} not supported by assertion comparison".format(type(a)))
+
+    # sometimes tests elevate warnings to errors -> make sure that does not happen here
+    with warnings.catch_warnings():
+        warnings.simplefilter("always")
+
+        assert type(a) == type(b)
+        if isinstance(a, (Variable, DataArray)):
+            assert a.equals(b), formatting.diff_array_repr(a, b, "equals")
+        elif isinstance(a, Dataset):
+            assert a.equals(b), formatting.diff_dataset_repr(a, b, "equals")
+        else:
+            raise TypeError("{} not supported by assertion comparison".format(type(a)))
```

Comment:
since we do this more than three times, always for the same reason *and* it increases the indentation depth (which is especially bad for `assert_allclose`, should we use a decorator instead?

Something like
```python
def ensure_warnings(func):
    @functools.wraps(func)
    def wrapper(*args, **kwargs):
        with warnings.catch_warnings():
            warnings.simplefilter("always")

            return func(*args, **kwargs)

    return wrapper
```